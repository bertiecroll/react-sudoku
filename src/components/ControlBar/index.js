export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
