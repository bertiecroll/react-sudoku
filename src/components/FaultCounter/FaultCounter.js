import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import './FaultCounter.css'
import { faultCount } from 'redux/selectors/fault'

export function FaultCounter ({ faultCount }) {
  return (
    <div className="FaultCounter">
      Faults: { faultCount === 0 ? 'none' : faultCount }
    </div>
  )
}

const mapStateToProps = state => ({ faultCount: faultCount(state) })
export default connect(mapStateToProps)(FaultCounter)

FaultCounter.defaultProps = {
  faultCount: 0,
}

FaultCounter.propTypes = {
  faultCount: PropTypes.number.isRequired,
}
