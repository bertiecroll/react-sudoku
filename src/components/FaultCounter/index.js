export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
