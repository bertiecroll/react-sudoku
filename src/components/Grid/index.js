export { Grid } from './Grid'
export { GridRow } from './GridRow'
