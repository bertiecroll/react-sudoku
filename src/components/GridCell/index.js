export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
