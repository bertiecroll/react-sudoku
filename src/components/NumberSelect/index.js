export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
