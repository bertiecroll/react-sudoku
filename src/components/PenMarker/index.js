export { PenMarker } from './PenMarker'
export { default } from './PenMarker'
