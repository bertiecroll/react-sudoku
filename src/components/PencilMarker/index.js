export { PencilMarker } from './PencilMarker'
export { default } from './PencilMarker'
