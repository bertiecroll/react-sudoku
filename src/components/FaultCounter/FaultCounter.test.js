import React from 'react'
import { shallow } from 'enzyme'
import { FaultCounter } from 'components/FaultCounter'

describe('FaultCounter', () => {
  it('renders a div containing text "none" when props.faultCount equals 0', () => {
    const wrapper = shallow(<FaultCounter faultCount={0} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a div containing faultCount when props.faultCount is greater than 0', () => {
    const wrapper = shallow(<FaultCounter faultCount={2} />)
    expect(wrapper).toMatchSnapshot()
  })
})
