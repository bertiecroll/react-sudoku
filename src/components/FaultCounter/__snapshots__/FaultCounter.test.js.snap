// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaultCounter renders a div containing faultCount when props.faultCount is greater than 0 1`] = `
<div
  className="FaultCounter"
>
  Faults: 
  2
</div>
`;

exports[`FaultCounter renders a div containing text "none" when props.faultCount equals 0 1`] = `
<div
  className="FaultCounter"
>
  Faults: 
  none
</div>
`;
